### ISSUE TITLE
Biome-specific music isn't working.

### ISSUE THREAD

**Issue by Duranson1** [telvarost/QuickAdditions-StationAPI#5]
Hi there, Trying to use your mod to play c418's 4 volume beta nether tracks, I have them in .ogg format with -nether-specific on the end of each of the 4 files in my custom-music folder, and now these tracks do play in the nether, but they play in the overworld as well.

**Comment by telvarost** [comment]
Hmm. Could you send me the full name of the song? The way the code works is that it trims the last three characters of the file name, and then checks if the file name ends with `-specific.` with `ogg` being trimmed out. It will then look for any tags that tell the song to play. If you are in level -1 (the nether) it will look for the `-nether-` tag.

Feel free to DM me on discord (you can find me on modification-station) or to continue to post images or responses here 👍

Example of some of the code which runs:
`            if (songName.endsWith("-specific.")) {`

**Comment by Duranson1** [comment]
Here are my four nether-specific music files in the "custom-music" directory:

ballad-of-the-cats-nether-specific.ogg
concrete-halls-nether-specific.ogg
dead-voxel-nether-specific.ogg
warmth-nether-specific.ogg

**Comment by telvarost** [comment]
Hmm, I just tested using the file name `ballad-of-the-cats-nether-specific.ogg`  and it worked.
Could it be that the song starts playing while you are in the nether and then you journey to the overworld?
The mod doesn't stop the song from playing when you switch dimensions, it only prevents it from starting if you are in the wrong dimension.

I still want to fix this bug for you if there is one. So any extra info you can give me to help recreate the issue is appreciated.
Maybe my mod is incompatible with another mod, or doesn't work on your current fabric version.
Send what info you can and I'll try to get to the bottom of it!

**Comment by Duranson1** [comment]
Nope, it plays randomly in the overworld unfortunately. I can just go ahead and link my modpack if that helps,  it's the exact instance that I'm running. https://modrinth.com/modpack/syncretic

**Comment by telvarost** [comment]
Still at work, but important to note that v2.1.0 of QuickAdditions incorrectly disabled mob despawning. Be sure to fix that config in your modpack! I'll take a look at the issue when I get off work or sometime tomorrow as today I may not have much time to work on it.

**Comment by Duranson1** [comment]
Thank you!! Just fixed that :) And no worries, no rush! Really appreciate your help

**Comment by Duranson1** [comment]
Just confirmed in game for sure that nether music will start playing at random in overworld.

**Comment by telvarost** [comment]
Could you message me on discord? I just tested your modpack, and the nether biome specific tag seems to be working properly. I can't seem to replicate your issue even with your modpack. I didn't even need to change the names of the songs.

You can do a similar test to see that its working. Remove all custom music tracks that aren't for the nether, then disable default Minecraft BGM and set the song interval to something very small like 10 for both min and max. Exit the game and reload. Wait in the overworld for 2-3 min. You'll notice no songs play. Enter the nether, a song will play immediately.
Add your custom songs back in and return to the overworld. Reload the game, one of the non-nether songs should play almost immediately. Add back the default Minecraft BGM.

I think what you are noticing is just that the songs take a bit to ramp up, so they may start in the nether but you don't really notice they are playing till a while after in the overworld.

**Comment by telvarost** [comment]
Maybe you mean to make a feature request to stop a song from playing when you exit the biome/dimension it is meant to play in?

Not a big fan of this for biomes, as the sound would cut out pretty harshly.
But, I guess it could make sense for dimensions because the switch has a loading bar and all

**Comment by telvarost** [comment]
I've created a new release of the mod that stops the song from playing if it is dimension specific and you exit that dimension. See the PR here: https://github.com/telvarost/QuickAdditions-StationAPI/pull/6

New version on Modrinth for it is v2.2.2:
https://modrinth.com/mod/quickadditions-stationapi/versions

**Comment by Duranson1** [comment]
I have confirmed that even in the overworld, Nether specific songs will start playing by themselves. I don't mind them continuing after exiting the nether, as that is an intended feature I'm just having trouble figuring out why they start in the overworld too. I'll go ahead and shoot you a message on Discord as well.

**Comment by telvarost** [comment]
As noted on discord, I have created a logging version of v2.2.2 so we can try to see what is happening on your machine. Let me know what the log output is when it tries to play a `-nether-specific.ogg` file with the `v2.2.2-logging` jar installed in your modpack:
https://github.com/telvarost/QuickAdditions-StationAPI/releases/tag/v2.2.2

**Comment by telvarost** [comment]
I have created a new version of code to try to combat this issue. Please let me know if v2.3.2 resolves this.

Thanks!

**Comment by Duranson1** [comment]
Finally works, thank you so much! Finally I have volume beta playing properly in my favorite version 😁

**Comment by telvarost** [comment]
Seems like this is working closing this issue now
